kbase-extension/static/kbase/js/common/cellComponents/tabs/jobStatus/jobStateList.js

```diff
@@ -92,47 +120,103 @@ define([
      */
     function factory(config) {
         const widgetsById = {},
-            logWidgets = {},
             bus = Runtime.make().bus(),
             { jobManager, toggleTab } = config;
 
         if (!jobManager.model || !jobManager.model.getItem('exec.jobs.byId')) {
             throw new Error('Cannot start JobStateList without a jobs object in the config');
         }
 
-        let container, tableBody, dropdownWidget;
+        let container, dropdownWidget, dataTable;
 
         /**
          * convert the table to a DataTable object to get sorting/paging functionality
          * @param {array} rows - table rows
          * @returns {DataTable} datatable object
          */
-        // Convert the table to a datatable object to get functionality
         function renderTable(rows) {
             const rowCount = rows.length;
-            return $(container.querySelector('table')).dataTable({
-                searching: false,
-                pageLength: dataTablePageLength,
-                lengthChange: false,
-                columnDefs: [
-                    {
-                        // action row should not be orderable
-                        targets: 2,
-                        orderable: false,
+            dataTable = $(container)
+                .find('table')
+                .dataTable({
+                    data: rows,
+                    rowId: (row) => {
+                        return 'job_' + row.job_id;
                     },
-                    {
-                        // logs row should not be orderable
-                        targets: 3,
-                        orderable: false,
+                    searching: false,
+                    pageLength: dataTablePageLength,
+                    lengthChange: false,
+                    columns: [
```

Comment:
each column in the table now has a function to render the column contents from the row data (the job state object)